src/common/hist_util.h

```diff
@@ -204,7 +204,7 @@ class HistCollection {
   }
 
   // have we computed a histogram for i-th node?
-  inline bool RowExists(bst_uint nid) const {
+  inline bool RowExists(bst_ulong nid) const {
```

Comment:
In general, we should stick with `bst_uint` for node id, as the fundamental tree type assumes that we have up to 2^{31} nodes.